homeassistant/components/light/mqtt_json.py

```diff
@@ -203,6 +210,16 @@ def state_received(topic, payload, qos):
                 except ValueError:
                     _LOGGER.warning("Invalid XY color value received")
 
+                try:
+                    hue = float(values['color']['h'])
+                    saturation = float(values['color']['s'])
```

Comment:
Personally, I think it might be good to validate the value ranges here (hue: 0-360, saturation: 0-100). The other color spaces don't do it either though, so I haven't added it here for consistency.